src/helpers.ts

```diff
@@ -73,6 +81,112 @@ export function createInterestRate(
+    return [null, createOrSetInitialEmptyIndexesAndRatesHistory(blockNumber, timestamp, market)];
+
+  const lastInvariant = _IndexesAndRatesHistory.load(
+    indexesUpdated ? market._previousIndexesAndRatesHistory! : market._lastIndexesAndRatesHistory
+  );
+  if (!indexesUpdated) indexesUpdated = new _IndexesAndRatesHistory(id);
+  if (!lastInvariant) log.critical("No last invariant", []);
+
+  const lastP2PSupplyIndex = lastInvariant!.newP2PSupplyIndex;
+  const lastP2PBorrowIndex = lastInvariant!.newP2PBorrowIndex;
+  const lastPoolSupplyIndex = market._reserveSupplyIndex;
+  const lastPoolBorrowIndex = market._reserveBorrowIndex;
+
+  const timestampDiff = timestamp.minus(lastInvariant!.timestamp);
+  const blockDiff = blockNumber.minus(lastInvariant!.blockNumber);
+
+  const proportionIdle = computeProportionIdle(
+    market._indexesOffset,
+    market._idleSupply,
+    market._p2pSupplyAmount,
+    lastP2PSupplyIndex
+  );
+  const supplyProportionDelta = computeProportionDelta(
+    market._p2pSupplyDelta,
+    market._p2pSupplyAmount,
+    lastPoolSupplyIndex,
+    lastP2PSupplyIndex,
+    proportionIdle,
+    __MATHS__
+  );
+  const borrowProportionDelta = computeProportionDelta(
+    market._p2pSupplyDelta,
+    market._p2pSupplyAmount,
+    lastPoolSupplyIndex,
+    lastP2PSupplyIndex,
+    proportionIdle,
+    __MATHS__
+  );
+
+  const newP2PSupplyRate = computeP2PSupplyRate(
+    market._poolBorrowRate,
+    market._poolSupplyRate,
+    market._p2pIndexCursor_BI,
+    market._reserveFactor_BI,
+    supplyProportionDelta,
+    proportionIdle,
+    __MATHS__
+  );
+
+  const newP2PBorrowRate = computeP2PBorrowRate(
+    market._poolBorrowRate,
+    market._poolSupplyRate,
+    market._p2pIndexCursor_BI,
+    market._reserveFactor_BI,
+    borrowProportionDelta,
+    proportionIdle,
+    __MATHS__
+  );
+
+  indexesUpdated.market = market.id;
+  indexesUpdated.blockNumber = blockNumber;
+  indexesUpdated.blockDiff = blockDiff;
+  indexesUpdated.timestamp = timestamp;
+  indexesUpdated.timestampDiff = timestampDiff;
+
+  indexesUpdated.lastP2PSupplyIndex = lastP2PSupplyIndex;
+  indexesUpdated.lastP2PBorrowIndex = lastP2PBorrowIndex;
+  indexesUpdated.lastPoolSupplyIndex = lastPoolSupplyIndex;
+  indexesUpdated.lastPoolBorrowIndex = lastPoolBorrowIndex;
+
+  indexesUpdated.newP2PSupplyRate = newP2PSupplyRate;
+  indexesUpdated.newP2PBorrowRate = newP2PBorrowRate;
+  indexesUpdated.newPoolSupplyRate = market._poolSupplyRate;
+  indexesUpdated.newPoolBorrowRate = market._poolBorrowRate;
+
+  indexesUpdated.newP2PSupplyIndex = __MATHS__.computeNewSupplyIndex(
+    lastP2PSupplyIndex,
+    newP2PSupplyRate,
+    timestampDiff,
+    blockDiff
+  );
+  indexesUpdated.newP2PBorrowIndex = __MATHS__.computeNewBorrowIndex(
+    lastP2PBorrowIndex,
+    newP2PBorrowRate,
+    timestampDiff,
+    blockDiff
+  );
+  indexesUpdated.newPoolSupplyIndex = market._reserveSupplyIndex;
+  indexesUpdated.newPoolBorrowIndex = market._reserveBorrowIndex;
```

Comment:
`createIndexesUpdated` can be call in a block where reserve is not updated. 
In this case, pool indexes are not the good ones